packages/backend/src/routers/admin.ts

```diff
@@ -122,4 +123,184 @@ export const adminRouter = t.router({
                 await ctx.env.kvDao.putProfessor(professor, true);
             }
         }),
+
+    // Audits
+    autoReportDuplicateUsers: protectedProcedure
+        .input(z.object({ cursor: z.string().optional() }).optional())
+        .mutation(async ({ ctx, input }) => {
+            // OpenAI Moderation API limits: 500 RPM, 10,000 RPD, 10,000 TPM
+            // Conservative batch size to stay well under limits
+            const BATCH_SIZE = 25; // Reduced further to account for moderation API limits
+            let duplicatesFound = 0;
+            let moderationFlagged = 0;
+            let processedCount = 0;
+            const MAX_HARASSMENT = 0.65;
+
+            // Get professors using cursor-based pagination
+            const profs = await ctx.env.kvDao.getAllProfessors();
+
+            // Find starting index based on cursor
+            let startIndex = 0;
+            if (input?.cursor) {
+                const cursorIndex = profs.findIndex((p) => p.id === input.cursor);
+                startIndex = cursorIndex >= 0 ? cursorIndex : 0;
+            }
+
+            // Process batch of professors
+            const endIndex = Math.min(startIndex + BATCH_SIZE, profs.length);
+            const batchProfessors = profs.slice(startIndex, endIndex);
+            const professorIds = batchProfessors.map((p) => p.id);
+
+            // Fetch professor data in smaller chunks to avoid KV limits
+            const FETCH_CHUNK_SIZE = 25; // Reduced chunk size for additional operations
+            const reportTasks: Promise<void>[] = [];
+            const professorUpdateTasks: Promise<unknown>[] = [];
+
+            for (let i = 0; i < professorIds.length; i += FETCH_CHUNK_SIZE) {
+                const chunk = professorIds.slice(i, i + FETCH_CHUNK_SIZE);
+
+                // eslint-disable-next-line no-await-in-loop
+                const professors = await ctx.env.kvDao.getBulkValues("professors", chunk);
+
+                // Process each professor in the chunk
+                for (const professor of professors) {
+                    // eslint-disable-next-line no-continue
+                    if (!professor) continue; // Skip null professors
```

Comment:
Would prefer a better comment here. Please explain how the allProfessors key could be out of sync if there is a concurrent request to the DB